pkg/webhook/pod/runtimeclass.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020-2021 Clastix Labs
+// SPDX-License-Identifier: Apache-2.0
+
+package pod
+
+import (
+	"context"
+	"net/http"
+
+	corev1 "k8s.io/api/core/v1"
+	nodev1 "k8s.io/api/node/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	capsulev1beta2 "github.com/clastix/capsule/api/v1beta2"
+	capsulewebhook "github.com/clastix/capsule/pkg/webhook"
+	"github.com/clastix/capsule/pkg/webhook/utils"
+)
+
+type runtimeClass struct{}
+
+func RuntimeClass() capsulewebhook.Handler {
+	return &runtimeClass{}
+}
+
+func (h *runtimeClass) class(ctx context.Context, c client.Client, name string) (client.Object, error) {
+	if len(name) == 0 {
+		return nil, nil
+	}
+
+	obj := &nodev1.RuntimeClass{}
+
+	if err := c.Get(ctx, types.NamespacedName{Name: name}, obj); err != nil {
+		if errors.IsNotFound(err) {
+			return nil, nil
+		}
```

Comment:
TIL that a Pod cannot be created using a non-existing Runtime Class: we can drop the `IsNotFound` error check and return it.

Reply:
You are right. will remove:

```
kubectl apply -f sample.yaml 
Error from server (Forbidden): error when creating "sample.yaml": pods "nginx" is forbidden: pod rejected: RuntimeClass "myclass" not found
```